zipline/assets/_assets.pyx

```diff
@@ -209,11 +203,19 @@ cdef class Equity(Asset):
         params = ', '.join(strings)
         return 'Equity(%d, %s)' % (self.sid, params)
 
+    @staticmethod
+    def from_dict(dict_):
```

Comment:
If we make this a class method, then we can implement it as:

```
@classmethod
def from_dict(cls, dict_):
    return cls(**dict_)
```
and we won't have to re-implement it in each subclass.